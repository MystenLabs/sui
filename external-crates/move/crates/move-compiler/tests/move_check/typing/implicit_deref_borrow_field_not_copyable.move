module 0x8675309::M {
    struct R has drop {}
    struct S has copy, drop {}
    struct B has drop { s: S, r: R }

    fun t1(b: B, bref: &B) {
        (b.s: S);
        R{} = b.r;

        (bref.s: S);
        R{} = bref.r;
    }

}
