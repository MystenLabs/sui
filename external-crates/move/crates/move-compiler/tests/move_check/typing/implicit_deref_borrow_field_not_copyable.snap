error[E05001]: ability constraint not satisfied
  ┌─ tests/move_check/typing/implicit_deref_borrow_field_not_copyable.move:8:15
  │
2 │     struct R has drop {}
  │            - To satisfy the constraint, the 'copy' ability would need to be added here
3 │     struct S has copy, drop {}
4 │     struct B has drop { s: S, r: R }
  │                                  - The type '0x8675309::M::R' does not have the ability 'copy'
  ·
8 │         R{} = b.r;
  │               ^^^ Invalid implicit copy of field 'r' without the 'copy' ability

error[E05001]: ability constraint not satisfied
   ┌─ tests/move_check/typing/implicit_deref_borrow_field_not_copyable.move:11:15
   │
 2 │     struct R has drop {}
   │            - To satisfy the constraint, the 'copy' ability would need to be added here
 3 │     struct S has copy, drop {}
 4 │     struct B has drop { s: S, r: R }
   │                                  - The type '0x8675309::M::R' does not have the ability 'copy'
   ·
11 │         R{} = bref.r;
   │               ^^^^^^ Invalid implicit copy of field 'r' without the 'copy' ability

